pkg/detector/policy.go

```diff
@@ -117,15 +118,14 @@ func (d *ResourceDetector) getAndApplyPolicy(object *unstructured.Unstructured,
 		return err
 	}
 
-	// Some resources are available in more than one group in the same kubernetes version.
-	// Therefore, the following scenarios occurs:
-	// In v1.21 kubernetes cluster, Ingress are available in both networking.k8s.io and extensions groups.
-	// When user creates an Ingress(networking.k8s.io/v1) and specifies a PropagationPolicy to propagate it
-	// to the member clusters, the detector will listen two resource creation events:
-	// Ingress(networking.k8s.io/v1) and Ingress(extensions/v1beta1). In order to prevent
-	// Ingress(extensions/v1beta1) from being propagated, we need to ignore it.
+	// if the updated object no longer match to the previously bound propagation policy, unbind them.
+	// in this case, we should remove the policy labels of resource and corresponding binding, and then directly return.
 	if !util.ResourceMatchSelectors(object, matchedPropagationPolicy.Spec.ResourceSelectors...) {
-		return nil
+		removeLabels := []string{
+			policyv1alpha1.PropagationPolicyNamespaceLabel,
```

Comment:
Does label `policyv1alpha1.PropagationPolicyPermanentIDLabel` should be deleted too?

Reply:
I've also thought about this. 

However, I finally decided to just keep the same logic as `cleanPPUnmatchedResourceBindings` function in `master` branch. You problem you said is another issue after all.

Reply:
If the PR of cleaning `policyv1alpha1.PropagationPolicyPermanentIDLabel` merged before my PR, I will rebase it and correct my this PR.

Reply:
you mean #4539?

Reply:
Is that PR only handle label cleaning when pp deleted? Is there any PR about handle label cleaning when pp updated? 

If not, we can create a PR to handle label cleaning when pp updated to focus on remove `policyv1alpha1.PropagationPolicyPermanentIDLabel` issue.

My this PR just keep the same logic as `cleanPPUnmatchedResourceBindings` with master branch.

Reply:
> Is that PR only handle label cleaning when pp deleted? Is there any PR about handle label cleaning when pp updated?
> 
yes
> If not, we can create a PR to handle label cleaning when pp updated to focus on remove `policyv1alpha1.PropagationPolicyPermanentIDLabel` issue.
> 
agree